sans/direct-beam.ipynb

```diff
@@ -0,0 +1,529 @@
+    "masks['promt-pulse'] = sc.less(tof['tof',1:], 1500.0 * sc.units.us) | \\\n",
+    "                      (sc.greater(tof['tof',:-1], 17500.0 * sc.units.us) & \\\n",
+    "                       sc.less(tof['tof',1:], 19000.0 * sc.units.us))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Note that this needs more tuning and masks too much. Better do this after moving detectors?\n",
+    "masks['beam-stop'] = sc.less(sc.abs(x), 0.03 * sc.units.m) \\\n",
+    "                   & sc.less(y, 0.028 * sc.units.m) \\\n",
+    "                   & sc.greater(y, -0.016 * sc.units.m)\n",
+    "pixel_counts.masks['beam-stop'] = sc.reshape(masks['beam-stop'], dims=pixel_counts.dims, shape=tuple(pixel_counts.shape))\n",
+    "plot(pixel_counts, vmax=1000, axes=['straw', 'tube', 'pixel'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "masks['tube-ends'] = sc.greater(x, 0.36 * sc.units.m) \\\n",
+    "                   | sc.less(x, -0.36 * sc.units.m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### X-Y projection\n",
+    "\n",
+    "Note: This cell can be skipped."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from scipp.plot import plot\n",
+    "plot(sans.project_xy(sc.sum(sample,'tof')))\n",
+    "plot(sans.project_xy(sample))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Instrument view\n",
+    "\n",
+    "We can use `groupby` to group pixels based on some criterion and then use the `copy` method to extract the desired group.\n",
+    "\n",
+    "#### Show single layer\n",
+    "\n",
+    "Note: This cell can be skipped."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from loki import LoKI\n",
+    "loki = LoKI()\n",
+    "sample.coords['layer'] = loki.layers()\n",
+    "loki.instrument_view(sc.groupby(sample, 'layer').copy(group=1), log=True)\n",
+    "del sample.coords['layer']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Show pixels with low number of counts\n",
+    "\n",
+    "Note: This cell can be skipped."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from loki import LoKI\n",
+    "loki = LoKI()\n",
+    "sample.coords['low-counts'] = sc.less(counts, 100.0*sc.units.counts)\n",
+    "loki.instrument_view(sc.groupby(sample, 'low-counts').copy(group=1))\n",
+    "del sample.coords['low-counts']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reduction and direct-beam iteration"
```

Comment:
What's being done below here is sufficiently complex that it could use some explanation for me. Though that is probably not required for our intended audience of scientists.